x-pack/plugins/observability_solution/infra/server/lib/alerting/metric_threshold/metric_threshold_executor.ts

```diff
@@ -324,12 +333,8 @@ export const createMetricThresholdExecutor =
           new Set([...(additionalContext.tags ?? []), ...options.rule.tags])
         );
 
-        const evaluationValues = alertResults.reduce((acc: Array<number | null>, result) => {
-          if (result[group]) {
-            acc.push(result[group].currentValue);
-          }
-          return acc;
-        }, []);
+        const evaluationValues = getEvaluationValues<Evaluation>(alertResults, group);
+        const thresholds = getThresholds<any>(criteria);
```

Comment:
can we use specific type here instead of `any`?

Reply:
It's been two weeks since I created this PR so it was a bit hard to remember 😄 I didn't find a proper type for this one because the params here have the following type:
```
export type MetricThresholdRuleParams = Record<string, any>;
```
But in the metric threshold rule, we had better typing. Please let me know if you are aware of a type that can be used here.